custom_code/views.py

```diff
@@ -1463,6 +1472,382 @@ def load_lightcurve_view(request):
     return HttpResponse(json.dumps(context), content_type='application/json')
 
 
+def load_dash_lightcurve_view(request):
+    """Lazy-load the dash lightcurve plot for the overview tab"""
+    target_id = request.GET.get('target_id')
+    width = int(request.GET.get('width', 600))
+    height = int(request.GET.get('height', 400))
+    
+    if target_id:
+        target = Target.objects.get(id=target_id)
+        # Create a context dict with request, as expected by dash_lightcurve
+        context_dict = {'request': request}
+        context = dash_lightcurve(context_dict, target, width, height)
+        html = render_to_string(
+            template_name='custom_code/dash_lightcurve.html',
+            context=context,
+            request=request
+        )
+        return JsonResponse({'plot_html': html}, safe=False)
+    return JsonResponse({'plot_html': '<div>Error loading plot</div>'}, safe=False)
+
+
+def load_spectra_plot_view(request):
+    """Lazy-load the spectra plot for the overview tab"""
+    target_id = request.GET.get('target_id')
+    
+    if target_id:
+        target = Target.objects.get(id=target_id)
+        # Create a context dict with request, as expected by spectra_plot
+        context_dict = {'request': request}
+        context = spectra_plot(context_dict, target)
+        html = render_to_string(
+            template_name='custom_code/spectra.html',
+            context=context,
+            request=request
+        )
+        return JsonResponse({'plot_html': html}, safe=False)
+    return JsonResponse({'plot_html': '<div>Error loading plot</div>'}, safe=False)
+
+
+def load_thumbnail_view(request):
```

Comment:
I think most views have a decorator saying if they are a get or a post. It looks like all of these are basically gets.